### PR TITLE
Fix nemo 1 packed sequence TE version error

### PR DESCRIPTION
**Pull request by cuichenx**
# What does this PR do ?

Adds a te version guard for packed sequence in NeMo 1 (NeMo 2 is unaffected)
Changes in #10688 should be paired with TE version >1.3
https://github.com/NVIDIA/TransformerEngine/blob/3d63cbb469ace2d1ad7798a956e16dee42bd655b/transformer_engine/pytorch/attention.py#L8179-L8185

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by xrennvidia** [comment]
Is this a safe hacking? wouldn't this impact functionality if you override `cu_seqlens_unpadded`?

**Comment by cuichenx** [comment]
> Is this a safe hacking? wouldn't this impact functionality if you override `cu_seqlens_unpadded`?

Tested in the case of CP=1 this is fine.  Not confident about CP>1. Do you have suggestions on how to tackle CP>1?

**Comment by xrennvidia** [comment]
I am thinking shouldn't we fix `PackedSeqParams` in NeMo2.0 and make it same as NeMo1.0 ?
@tomlifu , I remember you tested `cu_seqlens_unpadded != cu_seqlens` with TE < 1.13?

**Comment by tomlifu** [comment]
Yes, I tested cu_seqlens_unpadded != cu_seqlens with TE < 1.13 and it works. I don't think I used TE 1.13 at all.

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 1]
```diff
@@ -1366,7 +1367,9 @@ def fwd_output_and_loss_func(dataloader_iter, model, checkpoint_activations_all_
                             'attention_mask': None if self.get_attention_mask_from_fusion else batch['attention_mask'],
                             'labels': batch['labels'] if 'labels' in batch else None,
                         }
-
+                    if not is_te_min_version("1.13", check_equality=False):
```
Comment (tomlifu) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:1370]: @cuichenx I think you need an else here since this is only for the CP=1 case right? TE < 1.13 works for CP>1.

Comment (cuichenx) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:1370]: makes sense

**Comment by cuichenx** [comment]
> I am thinking shouldn't we fix `PackedSeqParams` in NeMo2.0 and make it same as NeMo1.0 ?
Certainly. The current PR just fixes a bug in nemo 1. We will propagate the change to nemo 2 in a future PR.

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!
